File: src/JSTestHost/ObjectModel/Payloads/TestRunRequestPayload.ts
import { TestRunSettings } from '../';
import { TestCase } from '../Common';

export interface StartExecutionWithSourcesPayload {
    TestRunSettings: TestRunSettings;
    Sources: Array<string>;
}

export interface StartExecutionWithTestsPayload {
    TestRunSettings: TestRunSettings;
    Tests: Array<TestCase>;
}

export interface StartDiscoveryPayload {
    TestRunSettings: TestRunSettings;
    Sources: Array<string>;
}

File: src/JSTest/Communication/Payloads/TestRunRequestPayload.cs
using Microsoft.VisualStudio.TestPlatform.ObjectModel;
using System;
using System.Collections.Generic;
using System.Text;

namespace JSTest.Communication.Payloads
{
	[Serializable]
    public class StartExecutionWithSourcesPayload
    {
        public TestRunSettings TestRunSettings;
        public IEnumerable<string> Sources;
    }

	[Serializable]
    public class StartExecutionWithTestsPayload
    {
        public TestRunSettings TestRunSettings;
        public IEnumerable<TestCase> Tests;
    }

    [Serializable]
    public class StartDiscoveryPayload
    {
        public TestRunSettings TestRunSettings;
        public IEnumerable<string> Sources;
    }
}
